services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2413,50 +2413,89 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
+        LOGGER.info(
+            "Data replication policy was not explicitly set when enabling active-active replication mode for store: {}."
+                + ". Setting it to ACTIVE_ACTIVE DRP.",
+            storeName);
+        updatedHybridStoreConfig.setDataReplicationPolicy(DataReplicationPolicy.ACTIVE_ACTIVE);
+        updatedConfigsList.add(DATA_REPLICATION_POLICY);
+      }
+
       // When turning off hybrid store, we will also turn off A/A store config.
       if (storeBeingConvertedToBatch && setStore.activeActiveReplicationEnabled) {
         setStore.activeActiveReplicationEnabled = false;
         updatedConfigsList.add(ACTIVE_ACTIVE_REPLICATION_ENABLED);
       }
 
-      // Update incremental push config.
-      setStore.incrementalPushEnabled =
-          incrementalPushEnabled.map(addToUpdatedConfigList(updatedConfigsList, INCREMENTAL_PUSH_ENABLED))
-              .orElseGet(currStore::isIncrementalPushEnabled);
-      // Enable incremental push automatically when batch user store being converted to hybrid store and active-active
-      // replication is enabled or being and the cluster config allows it.
-      if (!setStore.incrementalPushEnabled && !currStore.isSystemStore() && storeBeingConvertedToHybrid
-          && setStore.activeActiveReplicationEnabled
-          && clusterConfig.enabledIncrementalPushForHybridActiveActiveUserStores()) {
-        setStore.incrementalPushEnabled = true;
-        updatedConfigsList.add(INCREMENTAL_PUSH_ENABLED);
-      }
       // When turning off hybrid store, we will also turn off incremental store config.
       if (storeBeingConvertedToBatch && setStore.incrementalPushEnabled) {
         setStore.incrementalPushEnabled = false;
         updatedConfigsList.add(INCREMENTAL_PUSH_ENABLED);
       }
 
+      /*
+       * If active-active replication mode is being disabled, and store is still a hybrid store,
+       * the store is not a system store, then
+       * - change DRP ACTIVE_ACTIVE to NON_AGGREGATE DRP if hybridDataReplicationPolicy is not set
+       * - disable incremental push if it is enabled
+       */
+      if (activeActiveReplicationEnabled.isPresent() && !activeActiveReplicationEnabled.get()
+          && updatedHybridStoreConfig != null && !currStore.isSystemStore()) {
+        if (!hybridDataReplicationPolicy.isPresent()
+            && updatedHybridStoreConfig.getDataReplicationPolicy() == DataReplicationPolicy.ACTIVE_ACTIVE) {
+          LOGGER.info(
+              "Disabling active-active replication mode for store: {}. Setting data replication policy to NON_AGGREGATE",
+              storeName);
+          updatedHybridStoreConfig.setDataReplicationPolicy(DataReplicationPolicy.NON_AGGREGATE);
+          updatedConfigsList.add(DATA_REPLICATION_POLICY);
+        }
+        if (setStore.incrementalPushEnabled && clusterConfig.enabledIncrementalPushForHybridActiveActiveUserStores()) {
```

Comment:
I think if currentStore.incrementalPushEnabled is enough right?

Reply:
If we want to ensure that both AA and IncPush are fully dependent on one another, regardless of the `enabledIncrementalPushForHybridActiveActiveUserStores` configuration, then yes.

Reply:
Yeah we should ditch the clusterconfig, but what I want to say is should we look at currentStore, instead of setStore.

Reply:
IMO setStore.incrementalPushEnabled captures both new and old state
```
      setStore.incrementalPushEnabled =
          incrementalPushEnabled.map(addToUpdatedConfigList(updatedConfigsList, INCREMENTAL_PUSH_ENABLED))
              .orElseGet(currStore::isIncrementalPushEnabled);
```

Reply:
This code was removed.